server/dive_tasks/utils.py

```diff
@@ -157,3 +159,139 @@ def download_source_media(girder_client: GirderClient, datasetId: str, dest: Pat
         return [destination_path]
     else:
         raise Exception(f"unexpected metadata {str(dataset.dict())}")
+
+
+def upload_zipped_flat_media_files(
+    gc: GirderClient,
+    manager: JobManager,
+    folderId: str,
+    working_directory: Path,
+    create_subfolder=False,
+):
+    """Takes a flat folder of media files and/or annotation and generates a dataset from it"""
+    listOfFileNames = os.listdir(working_directory)
+    validation = gc.sendRestRequest('POST', '/dive_dataset/validate_files', json=listOfFileNames)
+    root_folderId = folderId
+    default_fps = gc.getFolder(root_folderId).get(f"meta.{constants.FPSMarker}", -1)
+    if validation.get('ok', False):
+        manager.write(f"Annotations: {validation['annotations']}\n")
+        manager.write(f"Media: {validation['media']}\n")
+        dataset_type = validation['type']
+        manager.write(f"Type: {dataset_type}\n")
+        if create_subfolder != '':
+            sub_folder = gc.createFolder(
+                folderId,
+                create_subfolder,
+                reuseExisting=True,
+            )
+            root_folderId = str(sub_folder["_id"])
+
+        # Upload all resulting items back into the root folder
+        manager.updateStatus(JobStatus.PUSHING_OUTPUT)
+        # create a source folder to place the zipFile inside of
+        gc.upload(f'{working_directory}/*', root_folderId)
+        if dataset_type == constants.ImageSequenceType and default_fps == -1:
+            default_fps = 1
+        gc.addMetadataToFolder(
+            str(root_folderId),
+            {constants.TypeMarker: dataset_type, constants.FPSMarker: default_fps},
+        )
+        # After uploading the default files we do a the postprocess for video conversion now
+        gc.sendRestRequest("POST", f"/dive_rpc/postprocess/{str(root_folderId)}")
+    else:
+        manager.write(f"Message: {validation['message']}\n")
+        manager.updateStatus(JobStatus.ERROR)
+
+
+def upload_exported_zipped_dataset(
```

Comment:
FWIW, this function appears to be quite fragile.  If we find that it breaks often, we may need to revisit this approach, but I think it's fine for now.